include/boost/math/special_functions/pow.hpp

```diff
@@ -46,7 +46,7 @@ template <int N>
 struct positive_power<N, 1>
 {
     template <typename T>
-    static T result(T base)
+    static constexpr T result(T base)
```

Comment:
These require C++14 constexpr anyway?

So I would change `constexpr` to `BOOST_CXX14_CONSTEXPR` and make the constexpr test conditional on BOOST_NO_CXX14_CONSTEXPR not being defined.  While it might be possible to rewrite these for C++11 constexpr it risks wrecking runtime performance.

Reply:
@jzmaddock Most functions in this header requires C++14 either due to variable declaration in the function body, or multiple return statements. I will proceed with the recommendation to change to BOOST_CXX_14_CONSTEXPR rather than bringing everything down to C++11 standard using obfuscated code.